src/backend/joanie/core/api/client.py

```diff
@@ -1380,3 +1381,70 @@ def preview_template(self, request, pk=None):  # pylint: disable=invalid-name, u
             as_attachment=True,
             filename="contract_definition_preview_template.pdf",
         )
+
+
+class NestedCourseOrderViewSet(NestedGenericViewSet, mixins.ListModelMixin):
+    """
+    Nested Order Viewset inside Course route.
+    It allows to list all learners who made orders that are 'validated' on a given course.
+    You may add some query parameters to filter out the list by organization or by product.
+
+    GET /api/courses/<course_id>/orders/
+        Returns every learners who made an order on a given course.
+
+    GET /api/courses/<course_id>/orders/?organization_id=<organization_id>>
+        Returns every learners who made an order on a course from a specific organization.
+
+    GET /api/courses/<course_id>/orders/?product_id=<product_id>
+        Returns every learners who made an order on the product's course.
+
+    GET /api/courses/<course_id>/orders/?organization_id=<organization_id>&product_id=<product_id>
+        Returns every learners that is attached to a product's course and an organization.
+    """
+
+    lookup_fields = ["order__course__pk"]
+    lookup_url_kwargs = ["course_id"]
+    ordering = ["-created_on"]
+    pagination_class = Pagination
+    permission_classes = [permissions.AccessPermission]
+    serializer_class = serializers.NestedCourseOrderSerializer
+    queryset = (
+        models.Order.objects.filter(state=enums.ORDER_STATE_VALIDATED)
+        .select_related(
+            "course",
+            "owner",
+            "organization",
+            "product",
+        )
+        .distinct()
+    )
+
+    def get_queryset(self):
+        """
+        Return the list of learners with a 'validated' order on a given course.
+        """
+        username = (
+            self.request.auth["username"]
+            if self.request.auth
+            else self.request.user.username
+        )
+        query_filters = {
+            "course_id": self.kwargs["course_id"],
+            "course__accesses__user__username": username,
+            "course__accesses__role__in": [
+                enums.ADMIN,
+                enums.INSTRUCTOR,
+                enums.MANAGER,
+                enums.MEMBER,
+                enums.OWNER,
+            ],
+        }
+        if self.request.query_params:
+            extra_query_filters = query_utility.generate_query_filters(
+                request=self.request,
+                url_query_parameters=["organization_id", "product_id"],
+                queryset_fields=["organization_id", "product_id"],
+            )
+            query_filters.update(extra_query_filters)
```

Comment:
I feel like this logic could be added to an existing mixin or in a new one, for re-usability.
WDYT ?

Reply:
To manage filters through query params, we are currently using [django-filter](https://pypi.org/project/django-filter/).

Take a look to https://github.com/openfun/joanie/blob/976cf3c490012052091f341de719d58759211068/src/backend/joanie/core/filters/client.py to know how it works.

Reply:
According to @jbpenrath feedback, I am taking away this part of the code and the method `generate_query_filters`.